pennylane_lightning/core/src/simulators/lightning_gpu/StateVectorCudaManaged.hpp

```diff
@@ -0,0 +1,1969 @@
+     *
+     */
+    void applyOperation_std(
+        const std::string &opName, const std::vector<size_t> &wires,
+        bool adjoint = false, const std::vector<Precision> &params = {0.0},
+        [[maybe_unused]] const std::vector<std::complex<Precision>>
+            &gate_matrix = {}) {
+        std::vector<CFP_t> matrix_cu(gate_matrix.size());
+        std::transform(gate_matrix.begin(), gate_matrix.end(),
+                       matrix_cu.begin(), [](const std::complex<Precision> &x) {
+                           return cuUtil::complexToCu<std::complex<Precision>>(
+                               x);
+                       });
+        applyOperation(opName, wires, adjoint, params, matrix_cu);
+    }
+
+    /**
+     * @brief Multi-op variant of `execute(const std::string &opName, const
+     std::vector<int> &wires, bool adjoint = false, const std::vector<Precision>
+     &params)`
+     *
+     * @param opNames
+     * @param wires
+     * @param adjoints
+     * @param params
+     */
+    void applyOperation(const std::vector<std::string> &opNames,
+                        const std::vector<std::vector<size_t>> &wires,
+                        const std::vector<bool> &adjoints,
+                        const std::vector<std::vector<Precision>> &params) {
+        PL_ABORT_IF(opNames.size() != wires.size(),
+                    "Incompatible number of ops and wires");
+        PL_ABORT_IF(opNames.size() != adjoints.size(),
+                    "Incompatible number of ops and adjoints");
+        const auto num_ops = opNames.size();
+        for (std::size_t op_idx = 0; op_idx < num_ops; op_idx++) {
+            applyOperation(opNames[op_idx], wires[op_idx], adjoints[op_idx],
+                           params[op_idx]);
+        }
+    }
+
+    /**
+     * @brief Multi-op variant of `execute(const std::string &opName, const
+     std::vector<int> &wires, bool adjoint = false, const std::vector<Precision>
+     &params)`
+     *
+     * @param opNames
+     * @param wires
+     * @param adjoints
+     * @param params
+     */
+    void applyOperation(const std::vector<std::string> &opNames,
+                        const std::vector<std::vector<size_t>> &wires,
+                        const std::vector<bool> &adjoints) {
+        PL_ABORT_IF(opNames.size() != wires.size(),
+                    "Incompatible number of ops and wires");
+        PL_ABORT_IF(opNames.size() != adjoints.size(),
+                    "Incompatible number of ops and adjoints");
+        const auto num_ops = opNames.size();
+        for (std::size_t op_idx = 0; op_idx < num_ops; op_idx++) {
+            applyOperation(opNames[op_idx], wires[op_idx], adjoints[op_idx]);
+        }
+    }
+
+    auto applyGenerator(const std::string &opName,
+                        const std::vector<size_t> &wires, bool adjoint = false)
+        -> PrecisionT {
+
+        return generator_map_.at(opName)(wires, adjoint);
```

Comment:
We may want a more safe approach to this call, in the event of a non-existent generator. Though, we can rectify this with support for arbitrary ops in Lightning.

Reply:
Check added.